### ISSUE TITLE
Don't we need AuthGuard on AccountController?

### ISSUE THREAD

**Issue by jlopezjuy** [jhipster/generator-jhipster-nodejs#175]
I have created a nre ms and i'm trying to register new user using swagger and i got 401 error.

```
import { Body, Param, Post, Res, UseGuards, Controller, Get, Logger, Req, UseInterceptors } from '@nestjs/common';
import { Response, Request } from 'express';
import { AuthGuard, RolesGuard } from '../../security';
import { User } from '../../domain/user.entity';
import { LoggingInterceptor } from '../../client/interceptors/logging.interceptor';
import { ApiBearerAuth, ApiUseTags, ApiResponse, ApiOperation } from '@nestjs/swagger';
import { AuthService } from '../../service/auth.service';

@Controller('api')
@UseInterceptors(LoggingInterceptor)
@UseGuards(AuthGuard, RolesGuard)
@ApiBearerAuth()
@ApiUseTags('account-resource')
export class AccountController {
  logger = new Logger('AccountController');

  constructor(private readonly authService: AuthService) {}
  @Post('/register')
  @ApiOperation({ title: 'Register user' })
  @ApiResponse({
    status: 201,
    description: 'Registered user',
    type: User
  })
  registerAccount(@Req() req: Request, @Body() user: User, @Res() res: Response): any {
    return res.sendStatus(201);
  }

  @Get('/activate')
  @ApiOperation({ title: 'Activate an account' })
  @ApiResponse({
    status: 200,
    description: 'activated'
  })
  activateAccount(@Param() key: string, @Res() res: Response): any {
    return res.sendStatus(200);
  }

  @Get('/authenticate')
  @ApiOperation({ title: 'Check if the user is authenticated' })
  @ApiResponse({
    status: 200,
    description: 'login authenticated'
  })
  isAuthenticated(@Req() req: Request): any {
    const user: any = req.user;
    return user.login;
  }

  @Get('/account')
  @ApiOperation({ title: 'Get the current user.' })
  @ApiResponse({
    status: 200,
    description: 'user retrieved'
  })
  async getAccount(@Req() req: Request): Promise<any> {
    const user: any = req.user;
    return await this.authService.findUserWithAuthById(user.id);
  }

  @Post('/account')
  @ApiOperation({ title: 'Update the current user information' })
  @ApiResponse({
    status: 201,
    description: 'user info updated',
    type: User
  })
  saveAccount(@Req() req: Request, @Body() user: User, @Res() res: Response): any {
    return res.sendStatus(201);
  }

  @Post('/account/change-password')
  @ApiOperation({ title: 'Change current password' })
  @ApiResponse({
    status: 201,
    description: 'user password changed',
    type: User
  })
  changePassword(@Req() req: Request, @Body() user: User, @Res() res: Response): any {
    return res.sendStatus(201);
  }

  @Post('/account/reset-password/init')
  @ApiOperation({ title: 'Send an email to reset the password of the user' })
  @ApiResponse({
    status: 201,
    description: 'mail to reset password sent',
    type: 'string'
  })
  requestPasswordReset(@Req() req: Request, @Body() email: string, @Res() res: Response): any {
    return res.sendStatus(201);
  }

  @Post('/account/reset-password/finish')
  @ApiOperation({ title: 'Finish to reset the password of the user' })
  @ApiResponse({
    status: 201,
    description: 'password reset',
    type: 'string'
  })
  finishPasswordReset(@Req() req: Request, @Body() keyAndPassword: string, @Res() res: Response): any {
    return res.sendStatus(201);
  }
}
```
i have removed @UseGuards(AuthGuard, RolesGuard) to register new account.

**Comment by amanganiello90** [comment]
Hi @jlopezjuy thanks for the interest! 
I know how is the controller, and it is right, you do not have to remove the Authorization Guard annotation, because you have to pass the token in the request in order to manage a user. In the swagger UI before you have to call the login api that returns the token. After pass it in the Authorization header with "bearer .xxxyourTokenxxx".

Then remember that now the user controller api is still not implemented (it creates, deletes or get anything and return always ok status code) for #166

**Comment by amanganiello90** [comment]
Resolved and implemented admin api in 1.4.0 release. Thanks
